src/components/microworld/microworld.jsx

```diff
@@ -0,0 +1,251 @@
+                     <Carousel items={all} />
+               </Box>
+            )
+        }
+        if (rows.length <= 0) {
+            return null;
+        }
+        return (
+            <div className="project-ideas">
+                <h1>Get inspiration from other projects</h1>
+                {rows}
+            </div>
+        )
+    },
+    renderForum: function() {
+        if (!this.state.microworld_data.show_forum) {
+            return null;
+        }
+
+        return (
+        <div className="forum">
+            <h1>Chat with others!</h1>
+            <img src="/images/forum-image.png"/>
+        </div>
+        )
+    },
+    renderDesignStudio: function() {
+        var designChallenge = this.state.microworld_data.design_challenge;
+        if (!designChallenge) {
+            return null;
+        }
+        if (designChallenge.studio_id) {
+            var studioHref = 'https://scratch.mit.edu//studios/' + designChallenge.studio_id + '/'
+        }
+        var designStudioIntro;
+        if (designChallenge.project_id) {
+            return (
+                <div className="side-by-side section">
+                    <h1>Join our Design Challenge!</h1>
+                    <div className="design-studio">
+                        <iframe src={"https://scratch.mit.edu/projects/" + designChallenge.project_id + "/#fullscreen"} frameBorder="0"> </iframe>
+                    </div>
+                    <div className="design-studio-projects">
+                        <Box title="Examples"
+                             key="scratch_design_studio"
+                             moreTitle={studioHref ? "Visit the studio" : null}
+                             moreHref={studioHref ? studioHref : null}>
+                            <Carousel settings={{slidesToShow:2,slidesToScroll:2}} items={this.state.microworld_data.design_challenge.studio1} />
+                            <Carousel settings={{slidesToShow:2,slidesToScroll:2}} items={this.state.microworld_data.design_challenge.studio2} />
+                        </Box>
+                    </div>
+                </div>
+            )
+        } else {
+            return ( 
+                <div className="section">
+                    <h1>Join our Design Challenge!</h1>
+                    <Box
+                        title="design Challenge Projects"
+                        key="scratch_design_studio"
+                        moreTitle={studioHref ? "Visit the studio" : null}
+                        moreHref={studioHref ? studioHref : null}>
+                        <Carousel items={this.state.microworld_data.design_challenge.studio1.concat(
+                            this.state.microworld_data.design_challenge.studio2)} />
+                   </Box>
+                </div>
+            )
+        }
+    },
+    render: function () {
+        var classes = classNames(
+            'top-banner'
+        );
+
+        var microworldData = this.state.microworld_data;
```

Comment:
You could probably use `this.props.microworld_data.<...>` down below, which means you could get rid of this assigment